src/main/java/com/openlattice/chronicle/services/twilio/TwilioService.kt

```diff
@@ -0,0 +1,63 @@
+package com.openlattice.chronicle.services.twilio
+
+import com.openlattice.chronicle.configuration.TwilioConfiguration
+import com.openlattice.chronicle.data.MessageDetails
+import com.openlattice.chronicle.constants.MessageOutcome
+import com.twilio.Twilio
+import com.twilio.exception.ApiException
+import com.twilio.rest.api.v2010.account.Message
+import com.twilio.type.PhoneNumber
+import org.slf4j.LoggerFactory
+import java.net.URI
+import java.time.OffsetDateTime
+
+/**
+ * @author toddbergman &lt;todd@openlattice.com&gt;
+ */
+class TwilioService(configuration: TwilioConfiguration) :
+    TwilioManager {
+    companion object {
+        protected val logger = LoggerFactory.getLogger(TwilioService::class.java)
+        private const val URL = "{{URL}}"
+    }
+
+    init {
+        Twilio.init(configuration.sid, configuration.token)
+    }
+
+    private val fromPhoneNumber: PhoneNumber = PhoneNumber(configuration.fromPhone)
+
+    private fun sendMessage(messageDetails: MessageDetails) :MessageOutcome {
+        val messageText = "Follow this link to enroll in Chronicle: {{URL}}".replace(URL, messageDetails.url)
```

Comment:
First, I don't think you need the `.replace`. You should be able to reference the url in the string:

```
val messageText = "Follow this link to enroll in Chronicle: ${messageDetails.url}"
```

No change required, just curious if the following works, i.e. sending a text formatted message with new lines.

```
val text = """
Follow this link to enroll in Chronicle:

${messageDetails.url}
""";
```

Reply:
I did not get an empty line when trying this.

Reply:
Damn. Thanks for trying!